src/Blueprint/Blueprint/Cmdlets/BlueprintArtifacts/GetAzureRMBlueprintArtifact.cs

```diff
@@ -0,0 +1,63 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Blueprint.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using System;
+using System.Management.Automation;
+using static Microsoft.Azure.Commands.Blueprint.Common.BlueprintConstants;
+
+namespace Microsoft.Azure.Commands.Blueprint.Cmdlets
+{
+    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "BlueprintArtifact", SupportsShouldProcess = true, DefaultParameterSetName = ParameterSetNames.ArtifactsByBlueprint), OutputType(typeof(PSBlueprintAssignment))]
+    public class GetAzureRmBlueprintArtifact : BlueprintArtifactsCmdletBase
+    {
+        #region Parameters
+        [Parameter(ParameterSetName = ParameterSetNames.ArtifactsByBlueprint, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.ArtifactName)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ArtifactsByBlueprint, Mandatory = true, ValueFromPipeline = true, HelpMessage = ParameterHelpMessages.BlueprintObject)]
+        [ValidateNotNull]
+        public PSBlueprintBase Blueprint { get; set; }
```

Comment:
We decided to start with object here since having an object will avoid asking the user for other parameters such as the MG id or the Subscription Id that the blueprints lives in. Requiring knowledge of where blueprint lives is prone to mistakes - user can provide wrong MG or Subscription id which will make the Get- fail early. In order to eliminate this early fail, we  decided to use the object. This blueprint object knows where it lives (MG or subscription) and also knows it's name so we can quickly retrieve the artifacts it contains.  For the GA though, we can add -BlueprintName with another parameter set.